odxtools/multiplexercase.py

```diff
@@ -71,7 +71,7 @@ def _mux_case_resolve_odxlinks(self, odxlinks: OdxLinkDatabase, *,
 
     def _resolve_snrefs(self, context: SnRefContext) -> None:
         if self.structure_snref:
-            ddds = odxrequire(context.diag_layer).diag_data_dictionary_spec
+            ddds = odxrequire(odxrequire(context.diag_layer).diag_data_dictionary_spec)
```

Comment:
why would a diag layer be none here?

Reply:
`diag_data_dictionary_spec` is optional according to the XSD, but so far it has been mandatory in `DiagLayer` objects...

Reply:
We are creating an empty spec when XML does not have one, right?

Reply:
> We are creating an empty spec when XML does not have one, right?

ok, maybe I misunderstood you here: currently, we indeed always create a DDDS object in the diag layer. after this PR, this is not the case for for ECU-SHARED-DATA layers. probably I should change that (good point, thanks!)

Reply:
changed in https://github.com/mercedes-benz/odxtools/pull/326/commits/d15213b31a0a24f4b124acf39baabcc744f7620e